common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt

```diff
@@ -64,33 +72,80 @@ class ContainerToolsRule(private val testInstance: Any) : TestRule {
         }
     }
 
-    private fun setUpRule() {
+    private fun setUpRule(description: Description) {
         ideaProjectTestFixture =
             IdeaTestFixtureFactory.getFixtureFactory().createLightFixtureBuilder().fixture
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.setUp() })
 
         MockKAnnotations.init(testInstance, relaxed = true)
         replaceServices()
+        createTestFiles(description.methodName)
     }
 
     private fun tearDownRule() {
+        filesToDelete.forEach { it.delete() }
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.tearDown() })
     }
 
     /**
      * Replaces all services annotated with [TestService].
      */
     private fun replaceServices() {
-        for (member in testInstance::class.declaredMemberProperties) {
-            if (member.annotations.filter { it is TestService }.isNotEmpty()) {
-                member as KProperty1<Any?, Any?>
-                member.isAccessible = true
-                val service: Any = member.get(testInstance)!!
-                setService(service)
+        for (member in getMembersWithAnnotation(TestService::class.java)) {
+            member as KProperty1<Any?, Any?>
+            member.isAccessible = true
+            val service: Any = member.get(testInstance)!!
+            setService(service)
+        }
+    }
+
+    /**
+     * Creates all [File] annotated with [TestFile] in the given directory name.
+     *
+     * @param directoryName the name of the directory to create the test files in
+     */
+    private fun createTestFiles(directoryName: String) {
+        for (member in getMembersWithAnnotation(TestFile::class.java)) {
+            member.isAccessible = true
+            if (!member.returnType.isSubtypeOf(File::class.createType()) ||
+                member !is KMutableProperty<*>
+            ) {
+                throw IllegalArgumentException(
+                    "@TestFile can only annotate mutable fields of type " +
+                        "java.io.File"
+                )
+            }
+
+            val annotation: TestFile? = member.findAnnotation()
+            val directory: File = FileUtil.createTempDirectory(directoryName, null)
```

Comment:
Will this directory be deleted on the test end?

Reply:
yes. that is managed by the platform